src/domains/ecommerce-app/06_functions.tf

```diff
@@ -0,0 +1,128 @@
+resource "azurerm_resource_group" "ecommerce_functions_rg" {
+  name     = format("%s-fn-rg", local.project)
+  location = var.location
+
+  tags = var.tags
+}
+
+# Subnet to host ecommerce reporting function
+module "ecommerce_reporting_functions_snet" {
+  source                                         = "git::https://github.com/pagopa/azurerm.git//subnet?ref=v4.3.2"
+  name                                           = "${local.project}-reporting-fn-snet"
+  address_prefixes                               = [var.cidr_subnet_ecommerce_functions]
+  resource_group_name                            = local.vnet_resource_group_name
+  virtual_network_name                           = data.azurerm_virtual_network.vnet.name
+  enforce_private_link_endpoint_network_policies = true
+
+  service_endpoints = [
+    "Microsoft.Web",
+  ]
+
+  delegation = {
+    name = "default"
+    service_delegation = {
+      name    = "Microsoft.Web/serverFarms"
+      actions = ["Microsoft.Network/virtualNetworks/subnets/action"]
+    }
+  }
+}
+
+module "ecommerce_reporting_function_app" {
+  source = "git::https://github.com/pagopa/azurerm.git//function_app?ref=v4.3.2"
+
+  resource_group_name = azurerm_resource_group.ecommerce_functions_rg.name
+  name                = "${local.project}-reporting-fn"
+  location            = var.location
+  health_check_path   = "info"
+  subnet_id           = module.ecommerce_reporting_functions_snet.id
+  runtime_version     = "~4"
+  os_type             = "linux"
+  linux_fx_version    = "Java|21"
+
+  always_on                                = var.ecommerce_function_always_on
+  application_insights_instrumentation_key = data.azurerm_application_insights.application_insights.instrumentation_key
+
+  app_service_plan_name = "${local.project}-plan-reporting-fn"
+  app_service_plan_info = {
+    kind                         = var.ecommerce_functions_app_sku.kind
+    sku_tier                     = var.ecommerce_functions_app_sku.sku_tier
+    sku_size                     = var.ecommerce_functions_app_sku.sku_size
+    maximum_elastic_worker_count = 0
+  }
+
+  storage_account_name = replace("pagopa-${var.env}-${var.location_short}-ecommtx-sa-fn", "-", "")
+
+  app_settings = {
+    FUNCTIONS_WORKER_RUNTIME       = "java"
+    FUNCTIONS_WORKER_PROCESS_COUNT = 4
+  }
+
+  allowed_subnets = [data.azurerm_subnet.apim_vnet.id]
+
+  allowed_ips = []
+
+  tags = var.tags
+}
+
+resource "azurerm_monitor_autoscale_setting" "ecommerce_reporting_function" {
```

Comment:
Maybe we can remove autoscaling given the nature of the function, even in production, we won’t need to scale.

Reply:
fixed in 28e483bd11f13693981253ff910884e95a7035e7